src/OpenLoco/src/Vehicles/VehicleHead.cpp

```diff
@@ -3651,11 +3651,60 @@ namespace OpenLoco::Vehicles
     }
 
     // 0x004AA625
-    void VehicleHead::sub_4AA625()
+    void VehicleHead::landCrashedUpdate()
     {
-        registers regs;
-        regs.esi = X86Pointer(this);
-        call(0x004AA625, regs);
+        VehicleBase* currentVehicle = reinterpret_cast<VehicleBase*>(this);
+        EntityId nextVehicleId = currentVehicle->id;
+
+        while (currentVehicle)
+        {
+            switch (currentVehicle->getSubType())
+            {
+                case VehicleEntityType::head:
+                    reinterpret_cast<VehicleHead*>(currentVehicle)->updateSegmentCrashed();
+                    break;
+                case VehicleEntityType::vehicle_1:
+                    // calls nullsub_21: empty subroutine
+                    break;
+                case VehicleEntityType::vehicle_2:
+                    // calls nullsub_22: empty subroutine
+                    break;
+                case VehicleEntityType::bogie:
+                    reinterpret_cast<VehicleBogie*>(currentVehicle)->updateSegmentCrashed();
```

Comment:
We are trying to move away from member functions and in general use free functions for most logic. Vehicle though was mostly worked on before we started doing this so it so far doesn't follow that convention. I would like to generally apply it though going forward (i.e. including this PR).

Reply:
As it would be a bit of a pain to rework this now going to leave this as is member function wise.